### PR TITLE
Prevent crash when store watchers have an empty watch event

### PR DESCRIPTION
**Pull request by amdprophet**
## What is this change?

Prevents an edge case, where watcher contents can be nil, from crashing a backend.

## Why is this change necessary?

Fixes https://github.com/sensu/sensu-go/issues/4176.

## Does your change need a Changelog entry?

Yes.

## Have you reviewed and updated the documentation for this change? Is new documentation required?

No new documentation is required.

## How did you verify this change?

I haven't seen the same 

## Is this change a patch?

No.

### CODE REVIEW

File: backend/schedulerd/check_watcher.go [hunk 1]
```diff
@@ -111,6 +111,10 @@ func (c *CheckWatcher) startWatcher() {
 	for {
 		select {
 		case watchEvent, ok := <-watchChan:
+			if watchEvent.CheckConfig == nil {
+				logger.Error("nil check config received from check config watcher")
+				return
+			}
```
Comment (echlebek) [inline backend/schedulerd/check_watcher.go:117]: This is probably not quite right, as it doesn't check the `ok` value before accessing `watchEvent.CheckConfig`

Comment (amdprophet) [reply backend/schedulerd/check_watcher.go:120]: Good catch. Fixed!

Comment (echlebek) [reply backend/schedulerd/check_watcher.go:120]: That's not quite right either, I think we need to stop the schedulers if this happens, like when `<-ctx.Done()`. :thinking:

Comment (echlebek) [reply backend/schedulerd/check_watcher.go:120]: Maybe put the stopping logic in a defer.

**Comment by amdprophet** [comment]
Closing in favour of https://github.com/sensu/sensu-go/pull/4248.

**Pull request closed without merging**